src/components/ReviewList.jsx

```diff
@@ -22,29 +24,49 @@ function ReviewList() {
       })
       .then((res) => {
         setReviews(res.data.reviews);
+      })
+      .catch((err) => {
+        setErr("404 category not found");
+      });
+
+    axios
+      .get(`https://tr-games-api.herokuapp.com/api/categories`)
+      .then((res) => {
+        setCategories(
+          res.data.categories.map((category) => {
+            return category.slug;
+          })
+        );
       });
   }, [category, sortColumn, sortOrder]);
 
   return (
-    <main className="reviewList">
-      <h2>Reviews</h2>
-      <Filter />
-      <SortBy setSortColumn={setSortColumn} setSortOrder={setSortOrder} />
-      <ul>
-        {reviews.map((review) => {
-          return (
-            <li key={review.review_id}>
-              <ReviewCard
-                review_id={review.review_id}
-                title={review.title}
-                category={review.category}
-                votes={review.votes}
-              />
-            </li>
-          );
-        })}
-      </ul>
-    </main>
+    <div>
+      {console.log(categories)}
+      {[...categories, undefined].includes(category) === false ? (
+        <p>{err}</p>
+      ) : (
```

Comment:
(make sure you remove your console.logs!)

I like the solution here but i think in code it's not as clear as it could be

Instead of fetching the categories from the api, mapping over them, storing them in state and *then* checking if the category exists in the render could you check straight after the api call and set an error similar to how you've handled errors elsewhere?

Reply:
Attempted this, can you check if it's any better? Feels the same amount of work just happening in a new place.